tests/handlers/test_receipts.py

```diff
@@ -286,6 +286,36 @@ def test_filters_out_receipt_event_with_only_hidden_receipt_and_ignores_rest(sel
             ],
         )
 
+    def test_handles_string_data(self):
```

Comment:
Could you add a docstring to this function that notes that this is unintended behaviour? i.e:

```python
def test_handles_string_data(self):
    """
    Tests that an invalid shape for read-receipts is handled.
    Context: https://github.com/matrix-org/synapse/issues/10603
```